packit_service/worker/testing_farm.py

```diff
@@ -126,27 +126,68 @@ def _artifact(
         return artifact
 
     def _payload(
-        self, chroot: str, artifact: Optional[Dict[str, Union[List[str], str]]] = None
+        self,
+        chroot: str,
+        artifact: Optional[Dict[str, Union[List[str], str]]] = None,
+        build_id: Optional[int] = None,
     ) -> dict:
-        """
+        """Prepare a Testing Farm request payload.
+
         Testing Farm API: https://testing-farm.gitlab.io/api/
 
         Currently we use the same secret to authenticate both,
         packit service (when sending request to testing farm)
         and testing farm (when sending notification to packit service's webhook).
         We might later use a different secret for those use cases.
 
+        Args:
+            chroot: Target TF chroot.
+            build_id: ID of the related copr build.
+            artifact: Optional artifacts, e.g. list of package NEVRAs
         """
         distro, arch = self.chroot2distro_arch(chroot)
         compose = self.distro2compose(distro, arch)
         fmf = {"url": self.fmf_url}
         if self.fmf_ref:
             fmf["ref"] = self.fmf_ref
 
+        if build_id is not None:
+            copr_build = CoprBuildModel.get_by_build_id(build_id)
+            build_log_url = copr_build.build_logs_url
+            srpm_build = copr_build.get_srpm_build()
+            srpm_url = srpm_build.url
+            nvr_data = copr_build.built_packages[0]
```

Comment:
Hm, `copr_build.built_packages` can apparently be None even if the `build_id` is specified, weird. Will try to debug what's going on tomorrow.

Reply:
From quickly looking into it, the problem can be we get the `copr_build` object from calling [`CoprBuildModel.get_by_build_id(build_id)`](https://github.com/packit/packit-service/blob/7accfee9b5a23583ca921616638ff067f2b5c6dc/packit_service/models.py#L940), but we don't specify the target, so the first matching `CoprBuildModel` is taken, which may not have set the `built_packages`. So the solution could be using directly `CoprBuildModel` object we already have or specify the target when  calling `get_by_build_id`.